pkg/util/ecs/common.go

```diff
@@ -8,151 +8,162 @@
+// UpdateContainerMetrics updates performance metrics for a list of internal
+// container representations based on stats collected from the ECS metadata v2 API
 func UpdateContainerMetrics(cList []*containers.Container) error {
 	for _, ctr := range cList {
-		stats, err := getContainerStats(ctr.ID)
+		stats, err := metadata.V2().GetContainerStats(ctr.ID)
 		if err != nil {
 			log.Debugf("Unable to get stats from ECS for container %s: %s", ctr.ID, err)
 			continue
 		}
+
+		stats.IO.ReadBytes = sumStats(stats.IO.BytesPerDeviceAndKind, "Read")
+		stats.IO.WriteBytes = sumStats(stats.IO.BytesPerDeviceAndKind, "Write")
+
 		// TODO: add metrics - complete for https://github.com/DataDog/datadog-process-agent/blob/970729924e6b2b6fe3a912b62657c297621723cc/checks/container_rt.go#L110-L128
 		// start with a hack (translate ecs stats to docker cgroup stuff)
 		// then support ecs stats natively
-		cpu, mem, io, memLimit := convertECSStats(stats)
+		cpu, mem, io, memLimit := convertMetaV2ContainerStats(stats)
 		ctr.CPU = &cpu
 		ctr.Memory = &mem
 		ctr.IO = &io
+
 		if ctr.MemLimit == 0 {
 			ctr.MemLimit = memLimit
 		}
 	}
 	return nil
 }
 
-// getECSContainers returns all containers exposed by the ECS API as plain ECSContainers
-func getECSContainers() ([]metadata.ContainerMetadata, error) {
-	meta, err := GetTaskMetadata()
-	if err != nil || len(meta.Containers) == 0 {
-		log.Errorf("Unable to retrieve task metadata")
-		return nil, err
+// convertMetaV2Container returns an internal container representation from an
+// ECS metadata v2 container object.
+func convertMetaV2Container(c v2.Container) *containers.Container {
+	container := &containers.Container{
+		Type:        "ECS",
+		ID:          c.DockerID,
+		EntityID:    containers.BuildTaggerEntityName(c.DockerID),
+		Name:        c.DockerName,
+		Image:       c.Image,
+		ImageID:     c.ImageID,
+		AddressList: parseContainerNetworkAddresses(c.Ports, c.Networks, c.DockerName),
 	}
-	return meta.Containers, nil
-}
 
-// getContainerStats retrives stats about a container from the ECS stats endpoint
-func getContainerStats(id string) (ContainerStats, error) {
-	var stats ContainerStats
-	client := http.Client{
-		Timeout: timeout,
+	createdAt, err := time.Parse(time.RFC3339, c.CreatedAt)
+	if err != nil {
+		log.Errorf("Unable to determine creation time for container %s - %s", c.DockerID, err)
+	} else {
+		container.Created = createdAt.Unix()
 	}
-	resp, err := client.Get(statsURL + "/" + id)
+	startedAt, err := time.Parse(time.RFC3339, c.StartedAt)
 	if err != nil {
-		return stats, err
+		log.Errorf("Unable to determine creation time for container %s - %s", c.DockerID, err)
+	} else {
+		container.StartedAt = startedAt.Unix()
 	}
-	defer resp.Body.Close()
 
-	decoder := json.NewDecoder(resp.Body)
-	err = decoder.Decode(&stats)
-	if err != nil {
-		return stats, err
+	if l, found := c.Limits["cpu"]; found && l > 0 {
+		container.CPULimit = float64(l)
+	} else {
+		container.CPULimit = 100
```

Comment:
just to confirm is it a percentage here? 
can we multiple cpu?

Reply:
My understanding is that this is expressed as a CPU share rather than a percentage.